File: python/side_effect/invoiceitemmanager.py
class InvoiceItemManager:
    def save_item_total_amount(self, quantity: float, price: float, tax_rate: float) -> None:
        total = quantity * price
        tax = total * tax_rate
        total_with_tax = total + tax
        
        with open("invoice_items.csv", "a") as f:
            f.write(f"{quantity};{price};{tax_rate};{total_with_tax}\n")

#After
class InvoiceItemManagerA:
    def save_item_total_amount(self, quantity: float, price: float, tax_rate: float) -> None:
        total = quantity * price
        tax = total * tax_rate
        total_with_tax = total + tax
        
        line = f"{quantity};{price};{tax_rate};{total_with_tax}\n"
        self.save(line)

    def save(self, line):
        with open("invoice_items.csv", "a") as f:
            f.write(line)

File: javascript/side_effect/invoiceitemmanager.ts
import { appendFileSync } from 'fs';

export class InvoiceItemManager {
    saveItemTotalAmount(quantity: number, price: number, taxRate: number): void {
      const total = quantity * price;
      const tax = total * taxRate;
      const totalWithTax = total + tax;
  
      const line = `${quantity};${price};${taxRate};${totalWithTax}\n`;
      appendFileSync('invoice_items.csv', line);
    }
  
  }
  
